TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/PruferTreeGenerator.java

```diff
@@ -0,0 +1,183 @@
+package org.opensearch.migrations;
+
+import lombok.Builder;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Random;
+import java.util.Stack;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * See https://en.wikipedia.org/wiki/Pr%C3%BCfer_sequence.
+ * @param <T>
+ */
+@Slf4j
+public class PruferTreeGenerator<T> {
+    @Builder
+    @ToString
+    private static class Link { int from; int to; }
+
+    public static class SimpleNode<T> {
+        public final T value;
+
+        public SimpleNode(T value) {
+            this.value = value;
+        }
+
+        List<SimpleNode<T>> children;
+        public boolean hasChildren() { return children != null && children.size() > 0; }
+        public Stream<SimpleNode<T>> getChildren() { return children == null ? Stream.of() : children.stream(); }
+    }
+
+    public interface NodeValueGenerator<T> {
+        T makeValue(int vertexNumber);
+    }
+
+    public interface Visitor<T> {
+        void pushTreeNode(SimpleNode<T> node);
+        void popTreeNode(SimpleNode<T> node);
+    }
+
+    public static void main(String[] args) {
```

Comment:
args aren't used at all, correct? So this has no configurable aspect around, e.g. how many nodes, correct? I'm fine with that, but also somewhat curious about the design choice. It's just not relevant for our current use case?

Reply:
I've converted this to a unit test in a separate file.